xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -426,6 +498,114 @@ private String processRouteConfig(RouteConfiguration config) {
     return null;
   }
 
+  /**
+   * Handles EDS response, which contains a list of ClusterLoadAssignment messages. Each
+   * ClusterLoadAssignment message contains information for endpoints in a cluster, grouped
+   * by localities. The response is NACKed if contains invalid information for gRPC's usage.
+   * Otherwise, an ACK request is sent to management server. Information in ClusterLoadAssignment
+   * messages are saved in local cache, in case of corresponding endpoint watchers are registered
+   * later. Endpoint watchers interested in endpoint information contained in this response are
+   * notified.
+   */
+  private void handleEdsResponse(DiscoveryResponse edsResponse) {
+    logger.log(Level.FINE, "Received an EDS response: {0}", edsResponse);
+    adsStream.edsRespNonce = edsResponse.getNonce();
```

Comment:
Better to avoid assigning a private field outside its class when possible, even it's inner class. You can assign it around the line 
https://github.com/grpc/grpc-java/pull/6370/files#diff-2ab746208e54edd7bcbba5fc408b6946R676

I also see `sendAckRequest()` is assigning this field again. Is `sendAckRequest()` the only right place to assign it?

Reply:
Fixed.

Update the saved nonce value as soon as receiving the `DiscoveryResponse`. Also, added a piece of comment to explain the logic behind.

Reply:
I noticed this pattern as well. The value passed as an argument is assigned back to the member...

Reply:
All places should have been fixed now. 😄